core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -299,9 +290,8 @@ public String toString() {
       Map<String, SnapshotRef> refs,
       List<StatisticsFile> statisticsFiles,
       List<PartitionStatisticsFile> partitionStatisticsFiles,
-      List<MetadataUpdate> changes,
-      boolean rowLineageEnabled,
-      long nextRowId) {
+      long nextRowId,
+      List<MetadataUpdate> changes) {
```

Comment:
Not that this PR also restores the convention that changes are passed last to the `TableMetadata` constructor. This is because changes are accumulated in `TableMetadata.Builder` and are not part of the metadata definition. I'd prefer not to mix changes into the actual table metadata.